internal/providers/msentraid/msentraid.go

```diff
@@ -26,6 +27,14 @@ func init() {
 
 const localGroupPrefix = "linux-"
 
+var expectedScopes = []string{
+	"openid",
+	"profile",
+	"email",
+	"GroupMember.Read.All",
+	"User.Read",
+}
```

Comment:
I don't want to spend too much effort on making this `CheckTokenScopes` provider agnostic, because we don't know if we can actually implement it for other providers (because we don't know if other providers set the extra `scope` field on the token). So I would prefer to keep it as is for now and refactor it once we support other providers and know if and how we can implement `CheckTokenScopes` for those.

Reply:
I really think from a testing perspective and API perspective, this allows for better readibility, avoiding doing duplication of the base scopes at least, and at worst, for other providers, this could be a no-op. So let’s try to not have a const duplicating the base scopes here.

Reply:
Done